### ISSUE TITLE
Flavors: Add `kramdown` flavor support

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#400]
## Problem

rumdl currently targets `standard` (CommonMark + common GFM extensions) plus a small set of documented flavors. `kramdown` is a widely used Markdown superset (notably the default renderer in Jekyll) and introduces syntax that can trigger false positives under `standard`.

## Request

Add a new flavor: `kramdown`, based on `standard`, with flavor-specific handling for kramdown-only constructs so they do not interfere with lint rules.

Configuration should match existing flavor behavior:

```toml
[global]
flavor = "kramdown"

[per-file-flavor]
"**/*.kramdown" = "kramdown"
```

Auto-detect `**/*.kramdown` as `kramdown` when no flavor is configured (keep `.md` and `.markdown` as `standard`).

## Syntax to support

At minimum, treat these as valid kramdown syntax (not HTML, not links, not stray text), so they do not cause spurious rule hits:

### Header IDs

```text
# Title {#my-id}
```

### Attribute List Definitions (ALD)

```text
{:ref-name: #myid .my-class}
```

### Block Inline Attribute Lists (block IAL)

```text
A paragraph.
{: #para-one}

> Quote
{:title="Quote title"}
{: #quote-id}
```

### Span Inline Attribute Lists (span IAL)

```text
This *is*{:.underline} `code`{:#id}{:.class}.
A [link](test.html){:rel='something'}.
```

### Definition lists

```text
Term
: Definition text
```

### Footnotes

```text
Text with a footnote.[^1]

[^1]: Footnote definition
```

### Extensions blocks (for example comments)

```text
{::comment}
This is ignored.
{:/comment}
```

## Acceptance criteria

- `flavor = "kramdown"` is accepted in config and documented in `docs/flavors.md`.
- New `docs/flavors/kramdown.md` describing supported constructs and any rule adjustments.
- Fixtures/tests covering the syntax above, asserting no false positives compared to `standard` for the affected rules.

## References

- https://kramdown.gettalong.org/syntax.html
- https://kramdown.gettalong.org/documentation.html
- https://kramdown.gettalong.org/quickref.html
- https://github.com/cabo/kramdown-rfc/blob/master/README.md

**Comment by rvben** [comment]
Thanks for the request, @pygarap. Implemented in v0.1.19. You can enable it with `flavor = "kramdown"` in your `.rumdl.toml`.
